compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -430,7 +430,7 @@ class TypeApplications(val self: Type) extends AnyVal {
    *  So hopefully our grand-children will not have to deal with this non-sense!
    */
   def translateJavaArrayElementType(using Context): Type =
-    if self.typeSymbol.isAbstractOrParamType && !self.derivesFrom(defn.ObjectClass) then
+    if self.typeSymbol.isAbstractOrParamType && (self.classSymbol eq defn.ObjectClass) then
```

Comment:
I don't understand how that condition matches the comment and what was there before. I'd like to see a more explanation what this is intended to express at this point.